src/abcdmicro/build_template.py

```diff
@@ -0,0 +1,428 @@
+            initial_template.get_metadata(), current_template.numpy()
+        ),
+    )
+
+
+# Helper function for multi-metric registration of a single subject
+def _register_subject_multimetric(
+    subject_volumes: dict[str, ANTsImage],
+    current_template: dict[str, ANTsImage],
+    weights: dict[str, float],
+) -> tuple[str, str, dict[str, ANTsImage]]:
+    """
+    Performs affine and SyN-only registration for a single subject against the
+    current template, returning the affine transform, transformation field and the warped images.
+    Args:
+        subject_volumes: A dictionary mapping modality names to ANTsImage objects for the subject.
+        current_template: A dictionary mapping modality names to ANTsImage objects for the current template.
+        weights: A dictionary mapping modality names to their corresponding weights for multivariate registration.
+    Returns:
+        A tuple containing:
+            - The file path to the affine transform (str).
+            - The file path to the warp field (str).
+            - A dictionary mapping modality names to their corresponding warped ANTsImage objects.
+    """
+
+    modalities = list(subject_volumes.keys())
+    primary_mod = modalities[0]
+
+    # Affine registration using only the main modality
+    affine_result = ants.registration(
+        fixed=current_template[primary_mod],
+        moving=subject_volumes[primary_mod],
+        type_of_transform="Affine",
+    )
+    affine_transform = affine_result["fwdtransforms"][-1]
+
+    # Setup multivariate metrics using remaining modalities
+    multivariate_metrics = []
+    for k in range(1, len(modalities)):
+        modality_k = modalities[k]
+
+        fixed_image = current_template[modality_k]
+        moving_image = subject_volumes[modality_k]
+
+        multivariate_metrics.append(
+            [
+                "mattes",
+                fixed_image,
+                moving_image,
+                weights[modality_k],
+                1,
+            ]
+        )
+
+    # Deformable registration (SyNOnly, initialized with affine)
+    deformable_result = ants.registration(
+        fixed=current_template[primary_mod],
+        moving=subject_volumes[primary_mod],
+        multivariate_extras=multivariate_metrics,
+        type_of_transform="SyNOnly",
+        initial_transform=affine_transform,
+    )
+
+    warp_field = deformable_result["fwdtransforms"][0]
+
+    # Warped image list (including the main one from the result)
+    warped_images = {primary_mod: deformable_result["warpedmovout"]}
+
+    # Apply transform to other modalities
+    for k in range(1, len(modalities)):
+        modality_k = modalities[k]
+        warped_images[modality_k] = ants.apply_transforms(
+            fixed=current_template[modality_k],
+            moving=subject_volumes[modality_k],
+            imagetype=1,
+            transformlist=[warp_field, affine_transform],
+            whichtoinvert=[0, 0],
+        )
```

Comment:
Is `imagetype=1` correct here? That's saying the volumes being operated on are vector volumes rather than scalar volumes so it may be a mistake but I am not fully clear on this from reading the `ants.apply_transforms` documentation.

(Elsewhere you set `imagetype=1` in a situation where what you're transforming is itself a deformation field, so it made sense to say it's a vector type image)